cypress/integration/e2e/atom.video.spec.js

```diff
@@ -0,0 +1,99 @@
+
+describe('Video', function () {
+    describe('Main media youtube video', function () {
+        beforeEach(function () {
+            disableCMP();
+        });
+        it('should render', function() {
+            cy.visit(`/Article?url=${mainMediaVideo}`);
+            cy.get(`#youtube-block-main-media-0`).should('be.visible');
+        })
+
+        it('should hide overlay on click', function() {
+            cy.get(`#youtube-overlay-S0CE1n-R3OY`).should('be.visible');
+            cy.get(`#youtube-overlay-S0CE1n-R3OY`).click();
+            cy.get(`#youtube-overlay-S0CE1n-R3OY`).should('not.be.visible');
+        })
+
+        it('should dispatch play to server', function () {
+            const listOfOphanVideoValues = []
+
+            cy.intercept({
+                url: /^http:\/\/ophan\.theguardian\.com\/img\/2/,
+                query: {
+                    viewId: /^\.*/,
+                    video: /^\.*/,
+                }
+            }, (req) => {
+                const url = new URL(req.url)
+                const videoValue = url.searchParams.get('video')
+                if (videoValue) listOfOphanVideoValues.push(JSON.parse(videoValue))
+            })
+
+            cy.visit(`/Article?url=${mainMediaVideo}`);
+            cy.get(`#youtube-block-main-media-0`).should('be.visible');
+
+
+            cy.get(`#youtube-overlay-S0CE1n-R3OY`).should('be.visible');
+            cy.get(`#youtube-overlay-S0CE1n-R3OY`).click();
+            cy.get(`#youtube-overlay-S0CE1n-R3OY`).should('not.be.visible');
+
+            // eslint-disable-next-line cypress/no-unnecessary-waiting
+            cy.wait(2000).then(function(){
+                expect(listOfOphanVideoValues).to.deep.include(mainMediaPlayEvent);
+            })
+
+        })
+    })
+
+    describe('Embed youtube video', function () {
+        it('should render', function() {
+            cy.visit(`/Article?url=${embedMediaVideo}`);
+            cy.get(`#youtube-block-41`).scrollIntoView();
+            cy.get(`#youtube-block-41`).should('be.visible');
+        })
+
+        it('should hide overlay on click', function() {
+            cy.get(`#youtube-overlay-N9Cgy-ke5-s`).should('be.visible');
+            cy.get(`#youtube-overlay-N9Cgy-ke5-s`).click();
+            cy.get(`#youtube-overlay-N9Cgy-ke5-s`).should('not.be.visible');
+        })
+
+        it('should dispatch play to server', function () {
+            const listOfOphanVideoValues = []
+            cy.intercept({
+                url: /^http:\/\/ophan\.theguardian\.com\/img\/2/,
+                query: {
+                    viewId: /^\.*/,
+                    video: /^\.*/,
+                }
+            }, (req) => {
+                const url = new URL(req.url)
+                const videoValue = url.searchParams.get('video')
+                if (videoValue) listOfOphanVideoValues.push(JSON.parse(videoValue))
+            })
```

Comment:
There might be a better way to write this. In other parts of the code we use this pattern:

```
                    cy.intercept('GET', '**/most-read-geo**', (req) => {
                        req.reply((res) => {
                            expect(res.body).to.have.property('heading');
                            expect(res.statusCode).to.be.equal(200);
                        });
                    });
```

Reply:
Note, this example is asserting on the response but think we should be mocking the response and only asserting on the request

Reply:
a lot of the complexity from these tests arise from `cy.intercept`. The issue with the way I currently set it up was that we catch all ophan events (yet we only want to assert 1). 

A true solution would be to make sure I only intercept the right API call, but I have had difficulty only matching that query. The URL is the same (so we can easily turn that into a string to make it look nicer) yet the queries are always a little different:
- viewId: randomly generated ID (so we match everything)
- video: `encodeURIComponent` of an object. Yet somehow it never matches when I try to match it. Here is what the object looks like:
`{"id":"gu-video-youtube-2b33a7b7-e639-4232-9ecd-0fb920fa8147","eventType":"video:content:play"}`